src/sas/qtgui/Utilities/Preferences/PreferencesWidget.py

```diff
@@ -1,12 +1,9 @@
-import logging
+import functools
 
-from PyQt5.QtWidgets import QComboBox, QDialog, QPushButton, QWidget, QLabel, QHBoxLayout, QVBoxLayout, QLineEdit, QCheckBox
-from typing import Optional, Any, List, Union, Callable
+from PyQt5.QtWidgets import QComboBox, QWidget, QLabel, QHBoxLayout, QVBoxLayout, QLineEdit, QCheckBox
+from typing import Optional, List, Union, Callable, Any
 
-from sas.system.config.config import config
-from sas.qtgui.Utilities.UI.PreferencesUI import Ui_preferencesUI
-
-logger = logging.getLogger(__name__)
+from sas.system import config
 
 
 def set_config_value(attr: str, value: Any):
```

Comment:
For this to work with the setter-callback-generator it also needs an optional dtype kwarg so that e.g. strings from textboxes can be cast to floats?  Unless there's another workaround for this.
```suggestion
def set_config_value(attr: str, value: Any, dtype: Optional[Any] = None):    
    """Helper method to set any config value, regardless if it exists or not
    :param attr: The configuration attribute that will be set
    :param value: The value the attribute will be set to. This could be a str, int, bool, a class instance, or any other
    :param dtype: The datatype to cast the input value to if casting is desired
    """
    if dtype is not None:
        config = dtype(config)
    setattr(config, attr, value)

```